app/models/reports/drug_stock_calculation.rb

```diff
@@ -88,21 +89,23 @@ def patient_days_calculation
       (stocks_on_hand.map { |stock| stock[:stock_on_hand] }.reduce(:+) / estimated_patients).floor
     end
 
-    def consumption_calculation(opening_balance, received, closing_balance)
+    def consumption_calculation(opening_balance, received, redistributed, closing_balance)
       return {consumed: nil} if [opening_balance, received, closing_balance].all?(&:nil?)
 
       {
         opening_balance: opening_balance,
         received: received,
+        redistributed: redistributed,
         closing_balance: closing_balance,
-        consumed: opening_balance + received - closing_balance
+        consumed: opening_balance + received - (redistributed || 0) - closing_balance
```

Comment:
This is the heart of the change here, right? I'm also a fan of the other method refactors going on in this class.

Reply:
Yes 👍🏼